cluster/saltbase/salt/kube-addons/initd

```diff
@@ -0,0 +1,123 @@
+#!/bin/bash
+#
+### BEGIN INIT INFO
+# Provides:   kube-addons
+# Required-Start:    $local_fs $network $syslog kube-apiserver
+# Required-Stop:
+# Default-Start:     2 3 4 5
+# Default-Stop:      0 1 6
+# Short-Description: Kubernetes Addon Object Manager
+# Description:
+#   Enforces installation of Kubernetes Addon Objects
+### END INIT INFO
+
+
+# PATH should only include /usr/* if it runs after the mountnfs.sh script
+PATH=/sbin:/usr/sbin:/bin:/usr/bin
+DESC="Kubernetes Addon Object Manager"
+NAME=kube-addons
+DAEMON_LOG_FILE=/var/log/$NAME.log
+PIDFILE=/var/run/$NAME.pid
+SCRIPTNAME=/etc/init.d/$NAME
+KUBECTL=/usr/local/bin/kubectl
+
+# Define LSB log_* functions.
+# Depend on lsb-base (>= 3.2-14) to ensure that this file is present
+# and status_of_proc is working.
+. /lib/lsb/init-functions
+
+function addon_manager_async() {
```

Comment:
The current monitoring setup does not rely on DNS. It relies on environment
variables for the purposes of service discovery.

On Tue, Jan 20, 2015 at 3:51 PM, Zach Loafman <notifications@github.com>
wrote:

> In cluster/saltbase/salt/kube-addons/initd
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/3601#discussion_r23267625>
> :
>
> > +
> > +# PATH should only include /usr/* if it runs after the mountnfs.sh script
> > +PATH=/sbin:/usr/sbin:/bin:/usr/bin
> > +DESC="Kubernetes Addon Object Manager"
> > +NAME=kube-addons
> > +DAEMON_LOG_FILE=/var/log/$NAME.log
> > +PIDFILE=/var/run/$NAME.pid
> > +SCRIPTNAME=/etc/init.d/$NAME
> > +KUBECTL=/usr/local/bin/kubectl
> > +
> > +# Define LSB log_* functions.
> > +# Depend on lsb-base (>= 3.2-14) to ensure that this file is present
> > +# and status_of_proc is working.
> > +. /lib/lsb/init-functions
> > +
> > +function addon_manager_async() {
>
> Also, this isn't actually true today. Existing kube-up.sh semantics call
> setup-monitoring, which create the cluster/addons/cluster-monitoring
> objects, then create DNS.
>
> I suspect what happens is that monitoring comes up and gropes around in
> the dark for DNS, and then eventually gets it and starts working. Which is
> fine. That's actually a reasonable approach for service finalization.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/3601/files#r23267625>
> .
>